### PR TITLE
C# Basic

### PR DESCRIPTION
**Pull request by Borshecs**
fix naming,

divided to separate files,

used CamelCase

### CODE REVIEW

File: ConsoleApp1/Example.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
+﻿using System;
+
+namespace ConsoleApp1
+
+{
+    interface IDateProvider
```
Comment (alexthvest) [inline ConsoleApp1/Example.cs:6]: DatA = данные
DatE = дата

File: Player/Program.cs [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        public void ShowInfo()
+        {
+            Console.WriteLine($"{GetType().Name} Damage: {Damage}"); 
+        }
+        
+    }
+
+    class Gun : Weapon
+    {
+        public override int Damage { get { return 10; } }
```
Comment (alexthvest) [inline Player/Program.cs:33]:
```suggestion
        public override int Damage => 10;
```

File: ConsoleApp1/Example.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    {
+        string GetDate();
+    }
+    interface IDateProcessor
```
Comment (jskonst) [inline ConsoleApp1/Example.cs:10]: Let's split to separate files

File: ConsoleApp1/Example.cs [hunk 4]
```diff
@@ -0,0 +1,56 @@
+﻿using System;
+
+namespace ConsoleApp1
```
Comment (jskonst) [inline ConsoleApp1/Example.cs:3]: Please - move this to CourseApp (or add stylecop and gitlab build steps here manually

File: Player/Program.cs [hunk 5]
```diff
@@ -0,0 +1,122 @@
+﻿using System;
```
Comment (jskonst) [inline Player/Program.cs:1]: Additionally - please - move  this to CourseApp

File: CourseApp.Tests/Sum3in2Re5.cs [hunk 6]
```diff
@@ -0,0 +1,19 @@
+namespace TestsSum
+{
+    using CourseApp;
+    using Xunit;
+
+    public class Sum_3in2_Re5
+    {
+        [Fact]
```
Comment (jskonst) [inline CourseApp.Tests/Sum3in2Re5.cs:8]: Please use inlineData to check multiple variants

File: CourseApp.Tests/Sum5in2Re3.cs [hunk 7]
```diff
@@ -0,0 +1,19 @@
+namespace TestsSum
+{
+    using CourseApp;
+    using Xunit;
+
+    public class Sum_5in2_Re3
```
Comment (jskonst) [inline CourseApp.Tests/Sum5in2Re3.cs:6]: tests for single method shoud be placed in 1 file in several methods 

just create CalculatorTests.cs (Or even Calculator.cs) in test package and place all code there

File: CourseApp/Func.cs [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+    public class Func
+    {
+        public double MathFunction(double x)
+        {
+            const double a = 2.25;
+            return Math.Pow(a, (x * x) - 1) - Math.Log((x * x) - 1, 10) + Math.Pow((x * x) - 1, 1 / 3);
+        }
+
+        public double[] TaskA(double xn, double xk, double dx)
```
Comment (jskonst) [inline CourseApp/Func.cs:13]: use List instead of array

File: CourseApp/Player/Player1.csproj [hunk 9]
```diff
@@ -0,0 +1,8 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
Comment (jskonst) [inline CourseApp/Player/Player1.csproj:1]: this one should be in same CourseApp, or configure stylecop

File: CourseApp/Player/Program.cs [hunk 10]
```diff
@@ -0,0 +1,122 @@
+﻿using System;
+
+namespace Player1
+{
+    interface IHasInfo
```
Comment (jskonst) [inline CourseApp/Player/Program.cs:5]: there is simple rule - 1 file - 1 class or 1 interface

**Pull request closed without merging**